### PR TITLE
ci: container improvements

### PR DESCRIPTION
**Pull request by kwvg**
## Additional Information

* This pull request pulls container-specific changes from [dash#6387](https://github.com/dashpay/dash/pull/6387), [dash#6400](https://github.com/dashpay/dash/pull/6400) and [dash#6421](https://github.com/dashpay/dash/pull/6421)

* The `HOST` check before running `setup_sdk.sh` isn't a part of the script itself as the script is written to be independent of external variables set. The caller is expected to know the conditions needed to run `setup_sdk.sh` as the script is _relatively_ agnostic to its environment.

* The `version` attribute in the [`develop`](https://github.com/dashpay/dash/blob/a8e2316d6f9c6726a498bcae2c5c5d7354769511/contrib/containers/develop/docker-compose.yml) and [`guix`](https://github.com/dashpay/dash/blob/a8e2316d6f9c6726a498bcae2c5c5d7354769511/contrib/containers/guix/docker-compose.yml) container's `docker-compose.yml` has been dropped as the attribute has been deprecated in the compose spec ([source](https://github.com/compose-spec/compose-spec/blob/65ef9f4a5d713b405a77c45c64263f2543e65267/spec.md#version-top-level-element-obsolete)).

* Using `LD_LIBRARY_PATH` to point to LLVM's libraries are acceptable and will not interfere with executing binaries built using the distro's packaged compiler as it will eventually search default paths and find the libraries shipped with the distro ([source](https://man7.org/linux/man-pages/man8/ld.so.8.html)).

* Currently, running LLDB will result in a "personality set failed: Operation not permitted" error ([source](https://discourse.llvm.org/t/running-lldb-in-a-container/76801)). This is caused by its attempt at disabling ASLR for debugging.

  To work around this error, the container will now operate under relaxed restrictions (`seccomp=unconfined`). As disabling ASLR is valuable when debugging and the container is meant for developers (i.e. it isn't used for CI), we have opted to relax restrictions instead of skipping ASLR disablement.

* As of `develop` (a8e2316d6f9c6726a498bcae2c5c5d7354769511), packages built by the container are stored in `/tmp`, which is inadvisable as it is the same directory used to store functional test runs and it's not too difficult to delete `/tmp`'s contents to save space in a long running [`develop`](https://github.com/dashpay/dash/blob/a8e2316d6f9c6726a498bcae2c5c5d7354769511/contrib/containers/develop/docker-compose.yml) container and then realize that both `shellcheck` and `cppcheck` are stored there and now you have to ditch the container you're working in and restart it.
  * To remedy this, packages are now built and stored in `/opt` in accordance with the FHS ([source](https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch03s13.html)). `/usr/local` was a contender but it's pre-populated, meanwhile `ls /opt` would give you a very quick picture of what's built for the container.

  * `/tmp` will not be entirely empty because [pypa/pip#10753](https://github.com/pypa/pip/issues/10753) results in residual `.pem` files leaking into `/tmp` and `pyenv` stores its build log there and keeping it around has some debug value.

## Breaking Changes

None expected.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests **(note: N/A)**
- [x] I have made corresponding changes to the documentation **(note: N/A)**
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: contrib/containers/guix/scripts/guix-start [hunk 1]
```diff
@@ -9,19 +9,10 @@ if [[ ! -d "${WORKSPACE_PATH}" || ! "${WORKSPACE_PATH}" = /* || ! -f "${WORKSPAC
...
 export SDK_PATH="${SDK_PATH:-${WORKSPACE_PATH}/depends/SDKs}"
+export SDK_SRCS="${SDK_PATH:-${WORKSPACE_PATH}/depends/sdk-sources}"
 
-# Check if macOS SDK is present, if not, download it
-if [[ ! -d "${SDK_PATH}/${XCODE_ARCHIVE}" ]]; then
-    echo "Preparing macOS SDK..."
-    mkdir -p "${SDK_PATH}"
-    curl -L "${XCODE_SOURCE}/${XCODE_ARCHIVE}.tar.gz" | tar -xz -C "${SDK_PATH}"
-fi
+setup-sdk
```
[the remark this answers is not in the document]
Comment (kwvg) [reply contrib/containers/guix/scripts/guix-start:15]: The script is being copied into `/usr/local/bin` (which is included in `PATH`, [source](https://github.com/dashpay/dash/blob/1ed017c0021025ea4396ea2960614c8047ff1a18/contrib/containers/guix/Dockerfile#L33)) as `setup-sdk` ([source](https://github.com/dashpay/dash/blob/1ed017c0021025ea4396ea2960614c8047ff1a18/contrib/containers/guix/Dockerfile#L84)). This has been alluded to in the warning comment in `setup-sdk.sh` ([source](https://github.com/dashpay/dash/blob/1ed017c0021025ea4396ea2960614c8047ff1a18/ci/dash/setup-sdk.sh#L8-L10)). The invocation here is going to be in the container's environment, so it is correct.

[the remark this answers is not in the document]
Comment (UdjinM6) [reply contrib/containers/guix/scripts/guix-start:15]: hmmm.. before this PR `contrib/containers/guix/scripts/guix-start` could be used outside of the container env to [build releases](https://github.com/dashpay/dash/blob/develop/doc/release-process.md#build-and-attest-to-build-outputs) and it's broken now

[the remark this answers is not in the document]
Comment (UdjinM6) [reply contrib/containers/guix/scripts/guix-start:15]: maybe rename `ci/dash/setup-sdk.sh` to `contrib/containers/guix/scripts/setup-sdk` and use that in CI instead?

[the remark this answers is not in the document]
Comment (kwvg) [reply contrib/containers/guix/scripts/guix-start]: Works for me, resolved in latest push!

**Review by UdjinM6 — APPROVED** [review]
Tested local macos build via docker-compose, Guix build on ubuntu via direct guix-start call and [GH Guix CI](https://github.com/UdjinM6/dash/actions/runs/12353504503). All looks good 👍 

ACK 04ce1fea52b392d61fa4cbcd6c9a934aa225879e

**Review by PastaPastaPasta — APPROVED** [review]
utACK 04ce1fea52b392d61fa4cbcd6c9a934aa225879e

**Pull request merged**